src/ts/components/pagination/pagination.tsx

```diff
@@ -1,235 +1,224 @@
-      return this.getMaxPages();
-    }
-
-    return page;
-  };
+  }, [itemCount, pageSize]);
 
-  private onClickPageNumber = (index: number, page: number) => {
-    const { currentPageNumber, changePage } = this.props;
+  const getMaxPages = React.useCallback(() => Math.ceil(numFullPages()), [
+    numFullPages,
+  ]);
 
-    if (currentPageNumber !== page && !this.getDisplayDots(index, page)) {
-      return () => changePage(this.getPageToGoTo(page, index));
-    }
-  };
+  const isNextButtonDisabled = React.useCallback(() => {
+    return !getMaxPages() || currentPageNumber === getMaxPages() || disabled;
+  }, [getMaxPages, currentPageNumber, disabled]);
 
-  private getStart = () => {
-    const { currentPageNumber } = this.props;
+  const isPrevButtonDisabled = React.useCallback(() => {
+    return currentPageNumber === 1 || disabled;
+  }, [currentPageNumber, disabled]);
 
+  const isPageButtonDisabled = React.useCallback(() => {
+    return itemCount <= pageSize || disabled;
+  }, [itemCount, pageSize, disabled]);
+
+  const shouldGetMorePages = React.useCallback(
+    () => getMaxPages() > MAX_BUTTONS,
+    [getMaxPages]
+  );
+
+  const getDisplayDots = React.useCallback(
+    (index: number, page: number) =>
+      shouldGetMorePages() &&
+      ((index === 1 && page > 2) ||
+        (index === MAX_BUTTONS - 2 && page < getMaxPages() - 1)),
+    [shouldGetMorePages, getMaxPages]
+  );
+
+  const getButtonType = React.useCallback(
+    (page: number, index: number) => {
+      if (currentPageNumber === page) {
+        return 'selected';
+      }
+      if (getDisplayDots(index, page)) {
+        return 'dots';
+      }
+
+      return undefined;
+    },
+    [currentPageNumber, getDisplayDots]
+  );
+
+  const getPageToGoTo = React.useCallback(
+    (page: number, index: number) => {
+      if (getMaxPages() > MAX_BUTTONS && index === 0 && page > 1) {
+        return 1;
+      } else if (
+        getMaxPages() > MAX_BUTTONS &&
+        index === MAX_BUTTONS - 1 &&
+        page < getMaxPages()
+      ) {
+        return getMaxPages();
+      }
+
+      return page;
+    },
+    [getMaxPages]
+  );
+
+  const onClickPageNumber = React.useCallback(
+    (index: number, page: number) => {
+      if (currentPageNumber !== page && !getDisplayDots(index, page)) {
+        return () => changePage(getPageToGoTo(page, index));
+      }
+    },
+    [currentPageNumber, changePage, getPageToGoTo, getDisplayDots]
+  );
+
+  const getStart = React.useCallback(() => {
     return Math.max(
       Math.min(
         currentPageNumber - LEFT_BUTTONS,
-        this.getMaxPages() - LEFT_BUTTONS - RIGHT_BUTTONS
+        getMaxPages() - LEFT_BUTTONS - RIGHT_BUTTONS
       ),
       1
     );
-  };
+  }, [currentPageNumber, getMaxPages]);
 
-  private getEnd = () =>
-    Math.min(this.getStart() + MAX_BUTTONS, this.getMaxPages() + 1);
-
-  private getRange = () => {
-    const { itemCount, pageSize } = this.props;
+  const getEnd = React.useCallback(
```

Comment:
This component feels like its hooks would benefit quite a lot from being broken out into separate files whilst you're here making big changes as its quite unweildy. Would serve as a nice example for others too :)